cangen.c

```diff
@@ -81,6 +81,10 @@
 #define MODE_FIX 2
 #define MODE_RANDOM_EVEN 3
 #define MODE_RANDOM_ODD 4
+#define MODE_FORMAT 5
```

Comment:
Cool improvement!

Two nitpicks:

`MODE_FORMAT` is very unspecific. What about `MODE_RANDOM_FIX` instead?

Reply:
Sounds good! I wasn't sure about naming, happy to change it. I used format because I think of it as a format string - maybe in the future, they may be more 'format string' supported but I don't have idea about anything additional useful now.

Reply:
MODE_FORMAT changed to MODE_RANDOM_FIX.